baker_cli.go

```diff
@@ -0,0 +1,181 @@
+package baker
+
+import (
+	"flag"
+	"fmt"
+	"html/template"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	flagHelpConfig = flag.String("help", "", "show help for a `component` (input/filter/output/upload) (use '*' to dump all)")
+	flagVersion    = flag.Bool("version", false, "print build version number")
+	flagVerbose    = flag.Bool("v", false, "verbose logging (debug level)")
+	flagQuiet      = flag.Bool("q", false, "quiet logging (warn level)")
+	flagPretty     = flag.Bool("pretty", false, "human-readable logging (unstructured logging)")
+	flagPProf      = flag.String("pprof", "", `run pprof server on host port provided (disabled if ""), use "localhost:"  for a free port`)
+)
+
+// Use `-ldflags="-X 'github.com/AdRoll/baker.BuildVersion=someversion'"` when building Baker to set this value
+var BuildVersion = "-- unknown --"
+
+// MainCLI provides a handy way to quickly create a Baker program only providing
+// the available Components.
```

Comment:
```suggestion
// MainCLI starts provides a handy way to quickly create a command-line interface to Baker 
// by providing the list of components available to build and run a topology.
```

Reply:
I'd add somewhere that `./baker topology.toml` runs the topology.

Reply:
@tommyblue also this hasn't been addressed yet
The first non positional CLI argument is going to be interpreted as the path to the topology file.